### ISSUE TITLE
Restarting too fast

### ISSUE THREAD

**Issue by Proclivis** [snoyberg/keter#165]
I have a problem where the application is rapidly starting over and over when it is coded for port 80, but works fine when coded for port 3000. The message from the app in the log file says it can't bind because the resource is busy. I checked and there are no processes running that would use port 80. However, the keter config file does not call out a port, so it probably defaults to 80.

I think my problem is probably conceptual. I assumed that both keter and the app would be configured for port 80, and somehow keter would manage connecting the app to the port. I'm starting to wonder if it acts like a proxy even when not configured with a stanza for reverse proxy to other servers.

How am I supposed to set this up? Should I be configuring keter for some other port than 80 and then the app on 80?

Here is the config file:

stanzas:
    - type: webapp
      exec: ../exec/host
      args: []
      hosts:
        - www.powercloud.world
      requires-secure: false
      
      extra-hosts:
        - "*.powercloud.world"
    - type: redirect
      hosts:
        - powercloud.world
      actions:
        - host: www.powercloud.world

**Comment by BartAdv** [comment]
Right, keter should bind to the 80 and proxy the stuff for you. Your app should listen on a port specified by the `PORT` env variable:

https://github.com/snoyberg/keter#bundling-your-app-for-keter

**Comment by creichert** [comment]
@BartAdv is right about using `PORT`. Is your app still re-starting rapidly when you use that?

**Comment by Proclivis** [comment]
When you say PORT, do you mean something like:

Code my Yesod app to use an ENV var called PORT, set it to something other than 80, and then configure the keter packet to set this ENV variable?

Or something else?

**Comment by Proclivis** [comment]
I found some info in the book, so I'll self serve...

**Comment by Proclivis** [comment]
For anyone else who tries this, the solution was to use this:

main = warpEnv MyApp

instead of:

main = warp 3000 MyApp

and this config file:

stanzas:
    - type: webapp
      exec: ../exec/host
      args: []
      env:
        FROM_KETER_CONFIG: PORT
      hosts:
        - www.powercloud.world
      requires-secure: false
      extra-hosts:
        - "*.powercloud.world"
    - type: redirect
      hosts:
        - powercloud.world
      actions:
        - host: www.powercloud.world
    - type: reverse-proxy
      reversed-host: www.powercloud.world
      reversed-port: 80
      reversing-host: localhost

If you are running without keter:

export PORT=3000
stack exec yourAppName

**Comment by jezen** [comment]
It looks like this is no longer an issue, so I'll close this.
